CHANGELOG.md

```diff
@@ -11,11 +11,24 @@ tag versions. The Bond compiler (`gbc`) and
 different versioning scheme, following the Haskell community's
 [package versioning policy](https://wiki.haskell.org/Package_versioning_policy).
 
+## Unreleased  ##
+* `gbc` & compiler library: TBD
+* IDL core version: TBD
+* IDL comm version: TBD
+* C++ version: TBD
+* C# NuGet version: TBD  (major bump needed)
+* C# Comm NuGet version: TBD
+
+### C# ###
+
+* **Breaking change** Reflection.IsBonded now recognizes custom IBonded
```

Comment:
I agree that some observable behavior has changed, which may cause some code somewhere to break. For similar fixes (like https://github.com/Microsoft/bond/issues/417) we've just considered these bug fixes. Am I missing some part of a larger picture?